src/libraries/System.IO.FileSystem/src/System/IO/EnumerationOptions.cs

```diff
@@ -96,6 +97,12 @@ internal static EnumerationOptions FromSearchOption(SearchOption searchOption)
         /// </remarks>
         public MatchCasing MatchCasing { get; set; }
 
+        /// <summary>
+        /// Stop recursion if current depth exceeds the value.
+        /// Default is int.MaxValue.
+        /// </summary>
```

Comment:
We could add in remarks about zero value too:
```
        /// <remarks>If <see cref="MaxRecursionDepth" /> is set to a negative number, the default value <see cref="int.MaxValue" /> is used. If <see cref="MaxRecursionDepth" /> is set to zero, returns the contents of the initial directory.</remarks>
```

Reply:
For sharing experience. In PowerShell repo we do the work async to exclude PR merge delaying. We create and reference in PR description new doc issue (or new PR in PowerShell-Docs repo) so that add new docs after functional PR merged.
This approach resolve follow issues:
1. English is not native for the contributor
2. The contributor can not know all rules of creating docs

Reply:
I'd change the second remark to something like:

`If <see cref="MaxRecursionDepth" /> is set to zero, enumeration returns the contents of the initial directory.`

Reply:
Done.